internal/repositories/equipment.go

```diff
@@ -484,3 +485,90 @@ func OptionalIntsPetKind(k []int64, field string) predicate.PetKind {
 		s.Where(sql.InInts(s.C(field), petKind...))
 	}
 }
+
+func (r *equipmentRepository) BlockEquipment(
+	ctx context.Context, id int, startDate, endDate time.Time, userID int) error {
+	tx, err := middlewares.TxFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	eqToBlock, err := tx.Equipment.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	// Get EquipmentStatusName form DB
+	eqStatusNotAvailable, err := tx.EquipmentStatusName.
+		Query().
+		Where(equipmentstatusname.Name(domain.EquipmentStatusNotAvailable)).
+		Only(ctx)
+	if err != nil {
+		return err
+	}
+
+	start, end, err := checkDates(&startDate, &endDate)
+	if err != nil {
+		return err
+	}
+
+	// Set a new EquipmentStatusName for current Equipment
+	_, err = eqToBlock.Update().SetCurrentStatus(eqStatusNotAvailable).Save(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Create a new EquipmentStatus and set startDate, endDate, Equipment and EquipmentStatusName
+	_, err = tx.EquipmentStatus.Create().
+		SetCreatedAt(time.Now()).
+		SetEndDate(*end).
+		SetStartDate(*start).
+		SetEquipments(eqToBlock).
+		SetEquipmentStatusName(eqStatusNotAvailable).
+		SetUpdatedAt(time.Now()).
+		Save(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Get OrderStatusName form DB
+	orStatusBlocked, err := tx.OrderStatusName.
+		Query().
+		Where(orderstatusname.Status(domain.OrderStatusBlocked)).
+		Only(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Find all Orders which have OrderStatusName booked and start from startDate and later
+	orders, err := eqToBlock.QueryOrder().
+		Where(order.RentStartGTE(*start), order.RentStartLTE(*end)). // rentStart must be in range of startDate..endDate
+		Where(order.RentEndGTE(*start)).                             // rentEnd must be equal or greater that startDate
+		Where(order.HasOrderStatusWith(orderstatus.
+			HasOrderStatusNameWith(orderstatusname.StatusEQ(domain.OrderStatusPrepared)))).
+		Where(order.HasOrderStatusWith(orderstatus.
+			HasOrderStatusNameWith(orderstatusname.StatusEQ(domain.OrderStatusApproved)))).
+		All(ctx)
+
+	// Set a new OrderStatusName for these Orders and create new OrderStatus for each Order
+	for _, order := range orders {
```

Comment:
Order has `current_status` field and this one should be updated manually. BTW, there is no `UpdateMany` function [yet](https://github.com/ent/ent/issues/1815).

Reply:
Fix: we can Update many by `Where`, it's correct now